theano/tensor/nnet/abstract_conv.py

```diff
@@ -177,6 +177,187 @@ def conv2d(input,
     return conv_op(input, filters)
 
 
+def deconv(input,
+           filters,
+           input_shape=None,
+           output_shape=None,
+           filter_shape=None,
+           border_mode='valid',
+           subsample=(1, 1),
+           filter_flip=True):
+    """This function will build the symbolic graph for deconvoloving a
+    mini-batch of a stack of 2D inputs with a set of 2D filters (using
+    gradient of convolution), such that it upsamples the input to the
+    desired output resolution.
```

Comment:
I think that naming them `conv2d_grad_wrt_inputs` and `conv2d_grad_wrt_weights` would be even clearer, but I would agree if you'd find them too long. Actually, to underline that it is not just the gradient of the conv2d op, it might be better to name them  `gradconv2d_wrt_inputs` and `gradconv2d_wrt_weights`.